pegen/pegen.c

```diff
@@ -262,6 +262,21 @@ expect_token(Parser *p, int type)
     return t;
 }
 
+Token *
+get_last_nonnwhitespace_token(Parser *p)
+{
+    int i = 1;
+    Token *token = NULL;
+    while (i <= p->mark) {
```

Comment:
Seems this can be formulated more readably using a `for` loop, e.g.
```
for (int m = p->mark - 1; m >= 0; m--) {
    token = p->tokens[m];
   ...
}
```